packages/webdriverio/src/context.ts

```diff
@@ -122,11 +124,15 @@ export class ContextManager {
             !this.#isNativeContext &&
             !this.#mobileContext
         ) {
-            const context = await this.#browser.getContext()
-            this.#mobileContext = typeof context === 'string' ?
-                context : typeof context === 'object' ?
-                    context.id :
-                    undefined
+            const context = await this.#browser.getContext().catch((err) => {
```

Comment:
@christian-bromann 

Do you also have the context of when this happens, this is not something that should occur when using Appium, unless they are for example 
- using wrong caps/WDIO gets back non W3C ones
- the driver is old as ...

Reply:
I think generally we can't make the assumption that just because a driver is considered as "mobile" they have the `getContext` command implemented. A driver can always fail to send the correct capabilities causing WebdriverIO to consider a session as mobile when it isn't. This patch will continue to log and elevate the problem but won't cause the whole session to fail.

What do you think?

Reply:
So my two-cents: 
We then might need to:

- Fix the isMobile check to make it more robust 
- Do this "fallback" for all commands

I also believe we can't "patch" the vendors (because this doesn't seem to be a specific Appium issue, unless a very old driver is used) if they don't return "valid" W3C responses.

Reply:
> * Fix the isMobile check to make it more robust

We should totally strive for that but unfortunately can't rely on vendors implementing the specifications correctly.

> * Do this "fallback" for all commands

What do you mean by "all commands"? If some one uses an unsupported command in their test, it should fail. However with these session manager (e.g. context, shadow root manager or the polyfill manager) we are enhancing WebdriverIO in the background and these things shouldn't cause the testrun to fail. Hence this patch which makes the context manager do nothing if `getContext` is not implemented.

> if they don't return "valid" W3C responses.

I think this is not the issue. The issue is that the command is not implemented by the vendor or WebDriver server in which case we should disable the context manager.

Reply:
> What do you mean by "all commands"? If some one uses an unsupported command
I mean for mobile commands specific commands. But I agree with what you say

> I think this is not the issue. The issue is that the command is not implemented by the vendor or WebDriver server in which case we should disable the context manager.
I believe you just fixed this "not implemented by WebDriver" with https://github.com/webdriverio/webdriverio/pull/13949, so I wonder if this "fix" is still needed. I would then still let it fail and throw a different error message (maybe throw the capabilities that are used) to determine which drivers are not supporting this. 
Regarding the vendors, from what I now see LT is doing it differently, they default to JSONWP

<img width="663" alt="image" src="https://github.com/user-attachments/assets/bdebc615-f4c6-4302-8a19-732dc7d5f44b">

and the W3C caps are not "Appium" proof

<img width="669" alt="image" src="https://github.com/user-attachments/assets/137e6081-6575-42d5-a717-34ce37ca55d3">

I'll put checking the vendors on my todo list so we can properly check if it's mobile based on the correct caps

Reply:
> I believe you just fixed this "not implemented by WebDriver" with #13949, so I wonder if this "fix" is still needed.

I put up this patch because I am afraid we run into more issues where our mobile capability detection fails and causes a failure with the whole test run. I think this patch makes sense no matter what because we just disable the context manager in case the command fails.

Reply:
> > I believe you just fixed this "not implemented by WebDriver" with #13949, so I wonder if this "fix" is still needed.
> 
> I put up this patch because I am afraid we run into more issues where our mobile capability detection fails and causes a failure with the whole test run. I think this patch makes sense no matter what because we just disable the context manager in case the command fails.

Then can you add extra logs for debugging purposes so we can update the logic that is behind this when we get issues?

Reply:
What would you suggest aside from the `log.warn(`Error getting context: ${err}`)`?

Reply:
Maybe also throw the requested capabilities so we can determine if the mobile checks have made an error

Reply:
done